### PR TITLE
Sequence positional weights for fine tuning

### PR DESCRIPTION
**Pull request by maximz**

### CODE REVIEW

File: biotransformers/lightning_utils/data.py [hunk 1]
```diff
@@ -151,14 +151,38 @@ def collate_fn(
...
+    # Do the same here: right-pad with 1s to make all weight arrays equal size.
+    maximum_weight_length = max(arr.shape[0] for arr in weight_arrays)
+    weight_matrix = np.array([np.pad(arr, (0, maximum_weight_length - arr.shape[0]), mode='constant', constant_values=1.) for arr in weight_arrays])
+
+    # Add weight=1 at start and/or end,
+    # based on alphabet.prepend_bos and alphabet.append_eos,
+    # which tell us whether the tokenizer adds tokens at the beginning or end of the sequence.
+    ones_vector = np.ones((weight_matrix.shape[0], 1))
+    to_hstack = (ones_vector if alphabet.prepend_bos else None, weight_matrix, ones_vector if alphabet.append_eos else None)
+    weight_matrix = np.hstack([mat for mat in to_hstack if mat is not None])
```
Comment (jmichuda) [inline biotransformers/lightning_utils/data.py:174]: see also the [pytorch-native way](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) of implementing, but this looks like it also works. not sure if there's a benefit to using pytorch functions over numpy.

also while i'm nitpicking engineering details...maybe also consider parameterizing the padding methodology.

File: biotransformers/lightning_utils/models.py [hunk 2]
```diff
@@ -64,20 +64,36 @@ def configure_optimizers(self) -> Tuple[List[torch.optim.Optimizer], List[Dict]]
...
+            logits_flatten,
+            targets_flatten,
+            # set reduction="none" to get loss for each instance,
+            # then do weighted sum ourselves
+            # reduction="sum",
+            reduction="none",
             ignore_index=self.alphabet.padding_idx,
         )
+        # Return weighted sum
+        return (cross_entropy_per_token * weights_flatten).sum()
```
Comment (jmichuda) [inline biotransformers/lightning_utils/models.py:86]: Nice and elegant implementation! I'd like to see a unit test or some dummy data to verify that it works as intended

**Review by jmichuda — COMMENTED** [review]
Awesome Maxim! Really quick turnaround here, too. Excited to see how the results look
